individual/tests/graphql_enrollment_test.py

```diff
@@ -0,0 +1,192 @@
+                    status: "{status}"
+                    customFilters: {custom_filters}
+                ) {{
+                {newline.join(self.ENROLLMENT_SUMMARY_KEYS)}
+                }}
+            }}'''
+
+            return self.query(
+                query_str,
+                headers={"HTTP_AUTHORIZATION": f"Bearer {self.admin_token}"}
+            )
+        
+        for i, case in enumerate(self.individual_enrollment_cases):
+            response = send_individual_enrollment_summary_query(case.benefit_plan.uuid, case.status, case.custom_filters)
+            self.assertResponseNoErrors(response)
+            summary = json.loads(response.content)['data']['individualEnrollmentSummary']
+            for (key, expected_val) in case.expected_summary.items():
+                self.assertTrue(key in summary.keys(), f"Expected summary to have key {key} but key not found")
+                self.assertTrue(
+                    summary[key] == expected_val,
+                    f'Expected test case {i} to have {key} value {expected_val}, but got {summary[key]}'
+                )
+    
+    def test_confirm_individual_enrollment(self):
+        def send_confirm_individual_enrollment_mutation(benefit_plan_id, status, custom_filters):
+            query_str = f'''
+                mutation {{
+                  confirmIndividualEnrollment(
+                    input: {{
+                      customFilters: {custom_filters},
+                      benefitPlanId: "{benefit_plan_id}",
+                      status: "{status}",
+                    }}
+                  ) {{
+                    clientMutationId
+                    internalId
+                  }}
+                }}
+            '''
+
+            return self.query(
+                query_str,
+                headers={"HTTP_AUTHORIZATION": f"Bearer {self.admin_token}"}
+            )
+        
+        SocialProtectionConfig.enable_maker_checker_logic_enrollment = False
+        successful_enrollments = []
+
+        for i, case in enumerate(self.individual_enrollment_cases):
+            expect_max_active_beneficiaries_exceeded = case.expected_summary["maxActiveBeneficiariesExceeded"]
+
+            response = send_confirm_individual_enrollment_mutation(case.benefit_plan.uuid, case.status, case.custom_filters)
+            self.assertResponseNoErrors(response)
+            content = json.loads(response.content)
+            id = content['data']['confirmIndividualEnrollment']['internalId']
+
+            if expect_max_active_beneficiaries_exceeded:
+                self.assert_mutation_error(id, _('mutation.max_active_beneficiaries_exceeded'), f"Case {i}")
+            else:
+                self.assert_mutation_success(id)
+                client_mutation_id = content['data']['confirmIndividualEnrollment']['clientMutationId']
+                wait_for_mutation(client_mutation_id)
+
+                if case.benefit_plan not in successful_enrollments:
```

Comment:
@weilu I want to improve this test. Currently I allow only one successful test case per programme, so that subsequent test cases don't have to adjust their expected numbers to account for previous successful enrollment test cases. But ideally I want each test case to run in isolation. 'Unenrolling' beneficiaries seems to be complex - my other idea was to let it pass through tasks and then deny all tasks, but keen to hear any other ideas